mergify_engine/rules/parser.py

```diff
@@ -45,6 +47,39 @@
     )
 )
 
+_day = (
+    pyparsing.Word(pyparsing.nums)
+    .setParseAction(lambda tokens: date.Day(int(tokens[0])))
+    .addCondition(
+        lambda tokens: tokens[0].value >= 1 and tokens[0].value <= 31,
+        message="day must be between 1 and 31",
+    )
+)
+_month = (
+    pyparsing.Word(pyparsing.nums)
+    .setParseAction(lambda tokens: date.Month(int(tokens[0])))
+    .addCondition(
+        lambda tokens: tokens[0].value >= 1 and tokens[0].value <= 12,
+        message="month must be between 1 and 12",
+    )
+)
+_year = (
+    pyparsing.Word(pyparsing.nums)
+    .setParseAction(lambda tokens: date.Year(int(tokens[0])))
+    .addCondition(
+        lambda tokens: tokens[0].value >= 1900 and tokens[0].value <= 9999,
+        message="year must be between 1900 and 9999",
+    )
+)
+_day_of_week = (
+    pyparsing.Word(pyparsing.nums)
+    .setParseAction(lambda tokens: date.DayOfWeek(int(tokens[0])))
+    .addCondition(
+        lambda tokens: tokens[0].value >= 1 and tokens[0].value <= 7,
+        message="day-of-week must be between 1 and 7",
```

Comment:
Can you check these messages are printed correctly?

Reply:
This message can show up only in tests. When we parse the configuration you never get the level of details. Pyparsing just prints the expression is expected. If this one fails, it tries the next one and returns a very long generic error if none matches. This error contains all possibilities.

Reply:
That's what I thought unfortunately 😢